src/Files.App/ViewModels/Properties/SecurityAdvancedViewModel.cs

```diff
@@ -1,218 +1,251 @@
-﻿using CommunityToolkit.Mvvm.ComponentModel;
-using CommunityToolkit.Mvvm.Input;
-using Files.App.Data.Items;
-using Files.App.Extensions;
-using Files.App.Filesystem;
+// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
 using Files.App.Filesystem.Security;
-using Files.App.Helpers;
 using Microsoft.UI.Xaml;
-using System.Threading.Tasks;
+using Vanara.PInvoke;
 using Windows.Storage;
 
 namespace Files.App.ViewModels.Properties
 {
 	public class SecurityAdvancedViewModel : ObservableObject
 	{
-		public ListedItem Item { get; }
+		private readonly PropertiesPageNavigationParameter _navigationParameter;
 
-		private readonly Window Window;
+		private readonly Window _window;
 
-		private AccessControlList _accessControlList;
-		public AccessControlList AccessControlList
+		private readonly string _path;
+
+		private readonly bool _isFolder;
+
+		public bool IsAddAccessControlEntryButtonEnabled =>
+			AccessControlList is not null &&
+			AccessControlList.IsValid;
+
+		public bool IsDeleteAccessControlEntryButtonEnabled =>
+			AccessControlList is not null &&
+			AccessControlList.IsValid &&
+			SelectedAccessControlEntry is not null &&
+			SelectedAccessControlEntry.IsInherited is false;
+
+		public IconFileInfo ShieldIconFileInfo { get; private set; }
+
+		public bool CurrentInstanceCanReadPermissions { get; private set; }
+
+		public bool CurrentInstanceCanChangePermissions { get; private set; }
+
+		public string DisableInheritanceOption
 		{
-			get => _accessControlList;
-			set
+			get
 			{
-				if (SetProperty(ref _accessControlList, value))
-				{
-					ChangeOwnerCommand.NotifyCanExecuteChanged();
-					AddAccessControlEntryCommand.NotifyCanExecuteChanged();
-					RemoveAccessControlEntryCommand.NotifyCanExecuteChanged();
-					DisableInheritanceCommand.NotifyCanExecuteChanged();
-					ReplaceChildPermissionsCommand.NotifyCanExecuteChanged();
-				}
+				//if (!_isProtected)
+				//	return "SecurityAdvancedInheritedEnable/Text".GetLocalizedResource();
+				//else if (_preserveInheritance)
+				//	return "SecurityAdvancedInheritedConvert/Text".GetLocalizedResource();
+				//else
+				//	return "SecurityAdvancedInheritedRemove/Text".GetLocalizedResource();
```

Comment:
Commented code, is there a reason to keep it? If so, add an explanation please.

Reply:
Disabled for now.